File: UI/wwwroot/Model/Docente_asignaturas.js
//@ts-check
// @ts-ignore
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
import { Asignaturas }  from './Asignaturas.js'
import { Docentes }  from './Docentes.js'
class Docente_asignaturas extends EntityClass {
   /** @param {Partial<Docente_asignaturas>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {Boolean}*/ Jefe;
   /**@type {String}*/ Observaciones;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Asignaturas} ManyToOne*/ Asignaturas;
   /**@type {Docentes} ManyToOne*/ Docentes;
}
export { Docente_asignaturas }

File: CAPA_NEGOCIO/Gestion_Cursos/Model/Docente_asignaturas.cs
using APPCORE;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Docente_asignaturas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id { get; set; }
       public int? Docente_id { get; set; }
       public int? Asignatura_id { get; set; }
       public bool? Jefe { get; set; }
       public string? Observaciones { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       [ManyToOne(TableName = "Asignaturas", KeyColumn = "Id", ForeignKeyColumn = "Asignatura_id")]
       public Asignaturas? Asignaturas { get; set; }
       [ManyToOne(TableName = "Docentes", KeyColumn = "Id", ForeignKeyColumn = "Docente_id")]
       public Docentes? Docentes { get; set; }
   }
}
